File: src/UI/sd-ui/src/api/leagues/requests/CreateLeagueRequest.js
// src/api/leagues/CreateLeagueRequest.js

export const buildCreateLeagueRequest = ({
  leagueName,
  description,
  pickType,
  tiebreaker,
  useConfidencePoints,
  rankingFilter,
  teamFilter,
  isPublic,
  dropLowWeeksCount
}) => {
  return {
    name: leagueName,
    description: description?.trim() || null,
    pickType: pickType.toLowerCase(),
    tiebreakerType: "totalPoints", // or dynamic if needed later
    tiebreakerTiePolicy: tiebreaker, // already string value like "earliest"
    rankingFilter: rankingFilter || null,
    conferenceSlugs: teamFilter,
    useConfidencePoints,
    isPublic,
    dropLowWeeksCount: dropLowWeeksCount || 0
  };
};

File: src/SportsData.Api/Application/UI/Leagues/Commands/CreateLeague/Dtos/CreateLeagueRequest.cs
namespace SportsData.Api.Application.UI.Leagues.Commands.CreateLeague.Dtos;

public class CreateLeagueRequest
{
    public required string Name { get; set; }

    public string? Description { get; set; }

    public required string PickType { get; set; } // "straight" or "spread" from UI

    public bool UseConfidencePoints { get; set; }

    public required string TiebreakerType { get; set; } // "none", "totalPoints"
    public required string TiebreakerTiePolicy { get; set; } // "earliest", etc.

    public string? RankingFilter { get; set; } // "AP_TOP_25", etc.

    public List<string> ConferenceSlugs { get; set; } = []; // slugs from UI

    public bool IsPublic { get; set; }

    public int? DropLowWeeksCount { get; set; }
}